### PR TITLE
Add endpoint to store scraped courses

### LINKED ISSUE #46: Add API endpoint to store scraped courses
## Objective
Add API to store scraped courses following the model of #53 .

* Useful comment: https://github.com/calpoly-csai/api/pull/53#issuecomment-590121657

## Key Result
Scraped courses are saved in database.

## Details

* Add API to accept a list of scraped courses
* Create `save_course()` function in `database_wrapper` to build `SQLAlchemy` `Courses` object and save it to the database.

### PR DESCRIPTION
**Pull request by austinsilveria**
## What's New?
* `save_courses` API accepts a JSON object containing a list of all scraped courses (`request.form` did not play nicely with lists)
* `courseId` is now a string to uniquely identify each course (e.g. `CSC 101`)
* `validate_input_keys()` pulled out to its own function to be reused

## Type of change (pick-one)
- [ ] Bug fix (_non-breaking change which fixes an issue_)
- [x] New feature (_non-breaking change which adds functionality_)
- [ ] Breaking change (_fix or feature that would cause existing functionality to not work as expected_)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Integration test with course scraper stores all scraped courses in local `MySQL` instance.

### Steps
1. Ensure your local connection to `MySQL`
2. Clone `https://github.com/calpoly-csai/csai-scraping`
3. Run `flask_api.py` in `api` package
4. Run `sustainer.py` in `csai-scraping` package (optionally comment out other scrapers if you do not want them to run) ([branch with updated `csai-scraping`](https://github.com/austinsilveria/course-web-scraper/tree/courseendpoint))

## Checklist (check-all-before-merge)
_formatting help: `- [x]` means "checked' and `- [ ]` means "unchecked"_

- [x] I documented my code according to the [Google Python Style Guide][1]

- [ ] I ran `./build_docs.sh` and the docs look fine

- [ ] I ran `./type_check.sh` and got no errors

- [x] I ran `./format.sh` because it automatically cleans my code for me 😄

- [x] I ran `./lint.sh` to check for what "format" missed

- [ ] I added my tests to the `/tests` directory

- [ ] I ran `./run_tests.sh` and all the tests pass

[1]: https://google.github.io/styleguide/pyguide.html

### CODE REVIEW

File: database_wrapper.py [hunk 1]
```diff
@@ -420,6 +423,32 @@ def save_audio_sample_meta_data(self, formatted_data: dict) -> bool:
 
         pass
 
+    def save_course(self, course_data: dict):
+        expected_keys = {'courseId', 'dept', 'courseNum', 'units',
```
Comment (mfekadu) [inline database_wrapper.py]: Does `save_course` need the `courseId` ? MySQL auto-increments the id field, right?

Comment (Jason-Ku) [reply database_wrapper.py]: Yeah if you have AUTO INCREMENT set you do not need to supply the `courseId`

File: Entity/Courses.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ class CollegeStanding(enum.Enum):
 class Courses(Base):
     __tablename__ = 'Courses'
     # TODO: update schema and this Courses class to follow snake_case convention
-    courseId = Column(Integer, primary_key=True)
+    courseId = Column(String(10), primary_key=True)
```
Comment (mfekadu) [inline Entity/Courses.py]: Let's undo this line because the primary key should be an integer!

**Review by mfekadu — CHANGES_REQUESTED** [review]

File: Entity/Courses.py (hunk 2 above)
Comment (mfekadu) [reply Entity/Courses.py]: good catch @Jason-Ku thank you!

File: database_wrapper.py (hunk 1 above)
Comment (austinsilveria) [reply database_wrapper.py]: Sounds good, `save_course` now uses auto incrementing id and checks if `Course` already exists by instead matching `dept` and `courseNum`

File: Entity/Courses.py (hunk 2 above)
Comment (austinsilveria) [reply Entity/Courses.py]: :thumbsup:

**Comment by mfekadu** [comment]
It worked for me and the code is fantastic, thank you @austinsilveria !

**Pull request merged**